python/argo/interaction.py

```diff
@@ -106,31 +134,46 @@ class Command(Interaction):
     corresponding command's appropriate representation.
     """
 
-    def _result_and_state(self) -> Tuple[Any, Any]:
+    def _result_and_state_and_out_err(self) -> Tuple[Any, Any, str, str]:
         res = self.raw_result()
         if 'error' in res:
             msg = res['error']['message']
             if 'data' in res['error']:
-                msg += " " + str(res['error']['data'])
+                msg += " " + str(res['error']['data']['data'])
             exception = ArgoException(res['error']['code'],
                                       msg,
-                                      res['error'].get('data'))
+                                      res['error'].get('data').get('data'),
+                                      res['error']['data']['stdout'],
+                                      res['error']['data']['stderr'])
             raise self.process_error(exception)
         elif 'result' in res:
-            return (res['result']['answer'], res['result']['state'])
+            return (res['result']['answer']['answer'],
+                    res['result']['answer']['state'],
+                    res['result']['stdout'],
+                    res['result']['stderr'])
         else:
             raise ValueError("Invalid result type from JSON RPC")
 
     def state(self) -> Any:
         """Return the protocol state after the command is complete."""
-        return self._result_and_state()[1]
+        return self._result_and_state_and_out_err()[1]
 
     def result(self) -> Any:
         """Return the result of the command."""
-        return self.process_result(self._result_and_state()[0])
+        return self.process_result(self._result_and_state_and_out_err()[0])
+
+    def stdout(self) -> str:
+        """Return the stdout printed during the execution of the command."""
```

Comment:
Isn't it the _contents_  of `stdout` being returned, rather than the file handle itself? (pardon my pedantry)